src/EnergyPlus/ThermalComfort.cc

```diff
@@ -2793,21 +2795,19 @@ namespace ThermalComfort {
                 }
                 state.dataThermalComforts->DailyAveOutTemp = 0.0;
 
-                auto epwFile = state.files.inputWeatherFilePath.open(state, "CalcThermalComfortAdaptiveASH55");
-                for (i = 1; i <= 8; ++i) { // Headers
-                    epwLine = epwFile.readLine().data;
-                }
+                WeatherManager::OpenEPlusWeatherFile(state, ErrorsFound, false);
```

Comment:
Is this change necessary?  It feels like it could have side effects, and is muddying up the changes here.  Maybe I'm misunderstanding why this is necessary.

Reply:
Do you refer to the change in the ASHRAE function while this pr is mainly about CEN15251? Or do you refer to the use of `OpenEPlusWeatherFile` in both functions?

Reply:
I'm wondering why you needed to change from

```
auto epwFile = state.files.inputWeatherFilePath.open(state, "CalcThermalComfortAdaptiveASH55");
```

to

```
WeatherManager::OpenEPlusWeatherFile(state, ErrorsFound, false);
```

I'm not actually saying the previous is correct, but I am wondering if that actually affects your bug fix here.  I'm trying to minimize the number of lines that your PR is changing to make sure we can get it reviewed and merged.  If this is an important change then it would be good to know why.  If it's not an important change, then it could potentially just make it tougher to review if it causes changes in results.

Reply:
Yes, I understand your point and totally agree with you, I was just trying to figure out which part should be altered. So let me put it this way.

These few lines involve skipping the epw header, and I currently changed that for both CEN and ASH. Changing of that for ASH to `OpenEPlusWeatherFile` is **not** essential for the sake of this pr, there was no error as to these specific lines (apart from some variable name changes in develop, but that was handled by pulling from upstream). I just did it to keep some consistency, as I was recommended to use `OpenEPlusWeatherFile` in CEN instead of hard coding the skipping.

In terms of `OpenEPlusWeatherFile`, I don't see any of its usage outside `WeatherManager.cc`, and there are hard-coded skipping instead. Maybe you want to do a standalone pr for that.

Reply:
My intent really is to just minimize the changes in this PR so that we can get this in for you.  I hate that this has taken so much effort and the amount of effort in reviewing goes up as the number of lines go up.  (Not always, but I think you get my point.)  

I would cut this down as far as possible just to the bare minimum set of changes so that we can look at the results changes and feel comfortable with them changing.

Reply:
Now no touching the ASH function, no use of  `OpenEPlusWeatherFile`, diffs should be much cleaner now. waiting to see ci runs.